application/cms/forms.py

```diff
@@ -92,12 +69,13 @@ def __call__(self, form, field):
 
 
 class DataSourceForm(FlaskForm):
-    remove_data_source = HiddenField()  # Updated via JS if a user wants to remove the data source
+    # Updated via JS if a user wants to remove the data source
+    remove_data_source = RDUCheckboxField(label="", choices=[("remove-source", "Remove source")])
```

Comment:
The `RDUCheckboxField` seems to assume that there are multiple choices, and so outputs the "Remove source" checkbox within a `<fieldset>` with an empty `<legend>`, which isn’t ideal accessibility-wise (as the un-described field set will get read out):

```
<div id="data-source-1-remove_data_source" name="data-source-1-remove_data_source" class="form-group js-hidden js-remove-data-source ">
  <fieldset >
    <legend> </legend>
          
<div class="multiple-choice">
  <input id="data-source-1-remove_data_source-0" name="data-source-1-remove_data_source" class="" value="remove-source"  type="checkbox">
  <label for="data-source-1-remove_data_source-0">Remove source</label>
</div>
      
  </fieldset>
</div>
```

The output could be simplified to:

```
<div class="multiple-choice js-hidden js-remove-data-source">
  <input id="data-source-1-remove_data_source-0" name="data-source-1-remove_data_source" class="" value="remove-source"  type="checkbox">
  <label for="data-source-1-remove_data_source-0">Remove source</label>
</div>
```

Could we update the `RDUCheckboxField` class to do this, if there's only one choice and/or no label?

Reply:
Yeah I wanted to tweak this to add a `RDUSingleCheckboxField`, but didn't end up finishing that. It wasn't a simple 5 minute job.

We should put in the effort to do this .. but it's probably not criminal to leave it as-is for now.

Reply:
I think I tested the simplified output you specified above, and it didn't style nicely, which is the point at which I gave up. But feel free to try again if you think you can get it under control. Ideally we would have something like:

```python
class RDUSingleCheckboxField(SelectField):
    widget = _RDUChoiceInput(type="checkbox")
```

We would then need to make sure that the `_choice_input.html` template fragment renders correctly when displayed standalone (i.e. a single checkbox by itself), as well as when it's part of a formgroup (e.g. `RDUCheckboxField`)

Reply:
Ok, commit e2152c66887c9b7fd9f14af6fc40f6636a1f2077 adds in the WIP bit. It seems to have a sticky ending - the title of the field below gets sucked up onto the end of the label.
![screen shot 2018-11-22 at 11 31 53](https://user-images.githubusercontent.com/2920760/48900417-3a6cce80-ee4a-11e8-998b-4e557d6ff85e.png)

Reply:
Note there's a bug with this that means it's taking in the wrong value (i.e. defaulting as checked ... investigating)